src/rand_impls.rs

```diff
@@ -148,8 +148,8 @@ macro_rules! float_impls {
         }
     }
 }
-float_impls! { f64_rand_impls, f64, 53, next_f64 }
-float_impls! { f32_rand_impls, f32, 24, next_f32 }
```

Comment:
Floats only have the number of mantissa bits as you corrected, and one implicit bit. But does the logic here not rely on that? Are you sure the `SCALE` calculated in the macro is correct now?

Reply:
The comment for `Closed01` is "rescale so that 1.0 - epsilon becomes 1.0 precisely". The rescaling is done by multiplying by `SCALE / (SCALE - 1.0)`. Now for `SCALE / (SCALE - 1.0)` to rescale `1.0 - EPSILON` into 1.0, we need `SCALE / (SCALE - 1.0) == 1.0 / (1.0 - EPSILON)`, which is true when `SCALE == 1.0 / EPSILON`. For `f32` ɛ is 2<sup>−23</sup>, so 1/ɛ (SCALE) should be equal to 1<<23, not 1<<24. For `f64` ɛ is 2<sup>−52</sup>, so 1/ɛ (SCALE) should be equal to 1<<52, not 1<<53.